### PR TITLE
Add Child Bounties to Polkadot and Kusama

### PR DESCRIPTION
**Pull request by shawntabrizi**

### CODE REVIEW

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -789,13 +789,29 @@ impl pallet_bounties::Config for Runtime {
...
+	type ChildBountyManager = ChildBounties;
 	type DataDepositPerByte = DataDepositPerByte;
 	type Event = Event;
 	type MaximumReasonLength = MaximumReasonLength;
 	type WeightInfo = weights::pallet_bounties::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub const MaxActiveChildBountyCount: u32 = 100;
+	pub const ChildBountyValueMinimum: Balance = BountyValueMinimum::get() / 10;
```
Comment (michalisFr) [inline runtime/kusama/src/lib.rs:801]: This is too high IMO. There may be use cases where the child bounty rewards need to be quite low ([example](https://docs.google.com/document/d/1V-39ktdgpvATPpV1cfI8iZlS8_3oBDAHdlcMDd_3jpo/edit?usp=sharing)) or at least disproportionally low to the total bounty.

I propose a fixed min of 5 * `DOLLARS` (half the min of the main bounty).

File: runtime/polkadot/src/lib.rs [hunk 2]
```diff
@@ -821,12 +822,28 @@ impl pallet_bounties::Config for Runtime {
...
-	type ChildBountyManager = ();
+	type ChildBountyManager = ChildBounties;
 	type DataDepositPerByte = DataDepositPerByte;
 	type MaximumReasonLength = MaximumReasonLength;
 	type WeightInfo = weights::pallet_bounties::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub const MaxActiveChildBountyCount: u32 = 100;
+	pub const ChildBountyValueMinimum: Balance = BountyValueMinimum::get() / 10;
```
Comment (michalisFr) [inline runtime/polkadot/src/lib.rs:833]: Same as for Kusama

This is too high IMO. There may be use cases where the child bounty rewards need to be quite low ([example](https://docs.google.com/document/d/1V-39ktdgpvATPpV1cfI8iZlS8_3oBDAHdlcMDd_3jpo/edit?usp=sharing)) or at least disproportionally low to the total bounty.

I propose a fixed min of 5 * `DOLLARS` (which I believe is half the min of the main bounty).

File: runtime/kusama/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:798]: will be addressed in follow up PR

File: runtime/polkadot/src/lib.rs (hunk 2 above)
Comment (shawntabrizi) [reply runtime/polkadot/src/lib.rs:829]: will be addressed in follow up PR

File: runtime/kusama/src/lib.rs [hunk 3]
```diff
@@ -1428,8 +1446,9 @@ construct_runtime! {
 		// Preimage registrar.
 		Preimage: pallet_preimage::{Pallet, Call, Storage, Event<T>} = 32,
 
-		// Bounties module.
+		// Bounties modules.
 		Bounties: pallet_bounties::{Pallet, Call, Storage, Event<T>} = 35,
+		ChildBounties: pallet_child_bounties = 40,
```
Comment (ggwpez) [inline runtime/kusama/src/lib.rs:1451]: We just pick a number that is not used yet?

**Review by ggwpez — APPROVED** [review]

File: runtime/kusama/src/lib.rs [hunk 4]
```diff
@@ -786,13 +786,29 @@ impl pallet_bounties::Config for Runtime {
...
+	type ChildBountyManager = ChildBounties;
 	type DataDepositPerByte = DataDepositPerByte;
 	type Event = Event;
 	type MaximumReasonLength = MaximumReasonLength;
 	type WeightInfo = weights::pallet_bounties::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub const MaxActiveChildBountyCount: u32 = 100;
+	pub const ChildBountyValueMinimum: Balance = BountyValueMinimum::get() / 10;
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:798]: no, but in this context, this is easier to write.

File: runtime/kusama/src/lib.rs (hunk 3 above)
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:1451]: Yes, each pallet needs a unique index for encoding into call enum

File: runtime/kusama/src/lib.rs [hunk 5]
```diff
@@ -786,13 +786,29 @@ impl pallet_bounties::Config for Runtime {
...
+	pub const ChildBountyCuratorDepositBase: Permill = Permill::from_percent(1);
+}
+
+impl pallet_child_bounties::Config for Runtime {
+	type Event = Event;
+	type MaxActiveChildBountyCount = MaxActiveChildBountyCount;
+	type ChildBountyValueMinimum = ChildBountyValueMinimum;
+	type ChildBountyCuratorDepositBase = ChildBountyCuratorDepositBase;
+	// TODO: UPDATE
+	type WeightInfo = pallet_child_bounties::weights::SubstrateWeight<Runtime>;
```
Comment (shawntabrizi) [inline runtime/kusama/src/lib.rs:809]:
```suggestion
	type WeightInfo = weights::pallet_child_bounties::WeightInfo<Runtime>;
```

**Review by emostov — APPROVED** [review]
Configuration looks reasonable, although I am not that familiar with the child bounties pallet

**Review by kianenigma — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**